source/v5/mqtt3_to_mqtt5_adapter.c

```diff
@@ -1819,12 +2393,26 @@ void aws_mqtt3_to_mqtt5_adapter_operation_table_remove_operation(
 
     struct aws_mqtt3_to_mqtt5_adapter_operation_base *operation = existing_element.value;
     if (operation != NULL) {
-        aws_mqtt3_to_mqtt5_adapter_operation_destroy(operation);
+        aws_mqtt3_to_mqtt5_adapter_operation_release(operation);
+    }
+}
+
+struct aws_mqtt3_to_mqtt5_adapter_operation_base *aws_mqtt3_to_mqtt5_adapter_operation_release(
+    struct aws_mqtt3_to_mqtt5_adapter_operation_base *operation) {
+    if (operation != NULL) {
+        aws_ref_count_release(&operation->ref_count);
     }
+
+    return NULL;
```

Comment:
Trivial: Is this required?

Reply:
It's the ref counting contract that I like since it allows you to minimally swap or drop temporary references without worrying about NULL.

In particular, assigning the result of acquire/release is always correct and safe.  If your release is in your destructor, then it doesn't add much, but if your release is mid-lifetime of an object, it simplifies the correct logic (you need to assign to NULL as well).